### ISSUE TITLE
Game over alert despite solvable position.

### ISSUE THREAD

**Issue by fwn** [Kiryhas/memechain#6]
First I want to note that I love your game and I play it regularly, sometimes even competitively on the same seed with my spouse. It's a great game!

Today the game over alert was triggered in still solvable position. Here's a screenshot:

![Screenshot_20230617-214830~2](https://github.com/Kiryhas/memechain/assets/7255296/c1d9df7d-a330-4409-9eb1-ff78dff6069a)

Now if I move one of the blue 32 boxes onto the other, the end of game alert box pops up:

![Screenshot_20230617-214914~2](https://github.com/Kiryhas/memechain/assets/7255296/e373d12d-9ad3-4380-9b41-097fb38210c4)

But once I press okay I can continue my game as usual:

![Screenshot_20230617-215955](https://github.com/Kiryhas/memechain/assets/7255296/840062ad-b5eb-4204-9be0-805707f7f420)

And it's in fact solvable, as you can see here:
![Screenshot_20230617-220114](https://github.com/Kiryhas/memechain/assets/7255296/9ffef7a1-bd05-4047-9a9d-6a413b913d86)

I can reproduce the error, even with the go back feature and, importantly, if I do collapse the light blue 2 onto the light blue 2 it stands on, the game over alert isn't triggered once I join the dark blue 32s.

It seems to have to do with the stack of light blue 2-stones, because when I go back to a position where there are two stone pairs still joinable:

![Screenshot_20230617-221003](https://github.com/Kiryhas/memechain/assets/7255296/409cfb91-4a4a-4ae1-a5a3-714f7faeaf2f)

I can join the dark blue 32s and reproduce the error with the pink 2s. If I join the light blue 2s first, no error occurs.

Anyway, tl;dr: fantastic game, but, you know.. literally unplayable.

**Comment by Kiryhas** [comment]
Hello!

I'm excited to know that you enjoy my game! I also want to thank you for your bug report. I expected such errors which is why reports such as yours are very important to improve the detection of combinations. Any chance you still have the game open and could run the following code and share the output?
```js
JSON.stringify(game.cubeManager.history)
```

If not, no problem. However, if you can, it would greatly speed up the debugging process.

Thanks!
